spyder/dependencies.py

```diff
@@ -307,9 +307,23 @@ def __init__(self, modname, package_name, features, required_version,
         self.required_version = required_version
         self.kind = kind
 
+        # Although this is not necessarily the case, it's customary that a
+        # package's distribution name be it's name on PyPI with hyphens
+        # replaced by underscores.
+        # Example:
+        # * Package name: python-lsp-black.
+        # * Distribution name: python_lsp_black
+        self.distribution_name = self.package_name.replace('-', '_')

```

Comment:
@ccordoba12, FWIW, `pkg_resources.get_distribution('python-lsp-server')` also works. I don't know if this is unique to python-lsp-server or if pkg_resources is smart enough to test both '-' and '_'.

Reply:
Also works for python-lsp-black, spyder-kernels, pyls-spyder, and diff-match-patch.

Reply:
Yep, I know. That's the custom among Python developers, but other packages could not follow this convention. So, it's just a heuristic that works for our deps, but every time we need to add a new dep that follows the `pylsp_black` case, we'll need to check if that's the case.